content/learn/book/migration-guides/0.8-0.9/_index.md

```diff
@@ -0,0 +1,638 @@
+```
+
+Converting “normal” parallel systems to exclusive systems is done by calling the exclusive ordering apis:
+
+```rust
+// Old (0.8)
+app.add_system(some_system.exclusive_system().at_end());
+// New (0.9)
+app.add_system(some_system.at_end());
+```
+
+Query state in exclusive systems can now be cached via ExclusiveSystemParams, which should be preferred for clarity and performance reasons:
+
+```rust
+// Old (0.8)
+fn some_system(world: &mut World) {
+  let mut transforms = world.query::<&Transform>();
+  for transform in transforms.iter(world) {
+  }
+}
+// New (0.9)
+fn some_system(world: &mut World, transforms: &mut QueryState<&Transform>) {
+  for transform in transforms.iter(world) {
+  }
+}
+```
+
+### [Merge TextureAtlas::from_grid_with_padding into TextureAtlas::from_grid through option arguments](https://github.com/bevyengine/bevy/pull/6057)
+
+`TextureAtlas::from_grid_with_padding` was merged into `from_grid` which takes two additional parameters for padding and an offset.
+
+```rust
+// 0.8
+TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1);
+// 0.9
+TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1, None, None)
+
+// 0.8
+TextureAtlas::from_grid_with_padding(texture_handle, Vec2::new(24.0, 24.0), 7, 1, Vec2::new(4.0, 4.0));
+// 0.9
+TextureAtlas::from_grid(texture_handle, Vec2::new(24.0, 24.0), 7, 1, Some(Vec2::new(4.0, 4.0)), None)
+```
+
+### [Spawn now takes a Bundle](https://github.com/bevyengine/bevy/pull/6054)
+
+```rust
+// Old (0.8):
+commands
+  .spawn()
+  .insert_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+commands.spawn_bundle((A, B, C));
+// New (0.9)
+commands.spawn((A, B, C));
+
+// Old (0.8):
+let entity = commands.spawn().id();
+// New (0.9)
+let entity = commands.spawn_empty().id();
+
+// Old (0.8)
+let entity = world.spawn().id();
+// New (0.9)
+let entity = world.spawn_empty();
+```
+
+### [Accept Bundles for insert and remove. Deprecate `insert`/`remove_bundle`](https://github.com/bevyengine/bevy/pull/6039)
+
+Replace `insert_bundle` with `insert`:
+
+```rust
+// Old (0.8)
+commands.spawn().insert_bundle(SomeBundle::default());
+// New (0.9)
+commands.spawn().insert(SomeBundle::default());
+```
+
+Replace `remove_bundle` with `remove`:
+
+```rust
+// Old (0.8)
+commands.entity(some_entity).remove_bundle::<SomeBundle>();
+// New (0.9)
+commands.entity(some_entity).remove::<SomeBundle>();
+```
+
+Replace `remove_bundle_intersection` with `remove_intersection`:
+
+```rust
+// Old (0.8)
+world.entity_mut(some_entity).remove_bundle_intersection::<SomeBundle>();
+// New (0.9)
+world.entity_mut(some_entity).remove_intersection::<SomeBundle>();
+```
+
+Consider consolidating as many operations as possible to improve ergonomics and cut down on archetype moves:
+
+```rust
+// Old (0.8)
+commands.spawn()
+  .insert_bundle(SomeBundle::default())
+  .insert(SomeComponent);
+
+// New (0.9) - Option 1
+commands.spawn().insert((

```

Comment:
Shouldn't it be 

```suggestion
commands.spawn_empty().insert((
```

Reply:
Yes, many of these migration guide sections were written prior to that change.